### PR TITLE
Separate server name

### PR DESCRIPTION
**Pull request by RonEld**
Seperate the server_name into the `server_addr`
which is the address to connect,
and to `server_name` which is the server host_name.
Sometimes these are not the same, and it helps separating them for connecting to other servers

### CODE REVIEW

File: tls-client/HelloHttpsClient.cpp [hunk 1]
```diff
@@ -70,15 +70,18 @@ const char *HelloHttpsClient::HTTP_HELLO_STR = "Hello world!";
...
                                    mbedtls_platform_context* in_platform_ctx) :
     socket(),
     server_name(in_server_name),
+    server_addr(in_server_addr),
     server_port(in_server_port),
     /* The platform context is passed just in case any crypto calls need it.
      * Please refer to https://github.com/ARMmbed/mbedtls/issues/1200 for more
      * information. */
     platform_ctx(in_platform_ctx)
+
```
Comment (andresag01) [inline tls-client/HelloHttpsClient.cpp]: Could you please remove this whitespace?

Comment (RonEld) [reply tls-client/HelloHttpsClient.cpp]: ok

**Comment by RonEld** [comment]
@andresag01 I agree that the examples should be simple, 
I have encountered several times that the server addr is different than the server name in the CN of the server's certificate. This application is used for reference and POC with different servers.
For example, if you use the exampleMbed TLS server on your PC (ssl_server2), and you want to connect your coard to your server, you will need a different server address ( your PC's IP \ computer name ) and server Name ( the CN of the default test certificate - localhost)
I have managed to connect to real servers ( MQTT servers in my case ), where the certificate CN was different than the server address.
If the hostname isn't equal to the Subject CN of the certificate, then certificate verification fails

**Comment by RonEld** [comment]
@andresag01 I remove the extra whitespace which was accidentally introduced in a merge conflict resolution

**Comment by andresag01** [comment]
All the relevant tests in the CI pass.

**Pull request merged**